File: java2csharp/java/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/ReversableLinearLayout.java
package com.samsung.android.sdk.professionalaudio.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

/**
 * This class is a LinearLayout which displays its children in reverse order when its orientation is
 * vertical and layoutdirection is RTL
 */
public class ReversableLinearLayout extends LinearLayout {

    public ReversableLinearLayout(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public ReversableLinearLayout(Context context) {
        super(context);
    }

    public ReversableLinearLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    /**
     * IMPORTANT
     * When the orientation is vertical and layoutdirection is RTL this method returns view counting
     * from the end. For example when index is 0 then last view is returnd.
     */
    @Override
    public View getChildAt(int index) {
        if (getOrientation() == LinearLayout.VERTICAL
                && getLayoutDirection() == ViewGroup.LAYOUT_DIRECTION_RTL) {
            index = getChildCount() - index - 1;
        }
        return super.getChildAt(index);
    }

}

File: java2csharp/csharp/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/ReversableLinearLayout.cs
namespace com.samsung.android.sdk.professionalaudio.widgets
{

	using Context = android.content.Context;
	using AttributeSet = android.util.AttributeSet;
	using View = android.view.View;
	using ViewGroup = android.view.ViewGroup;
	using LinearLayout = android.widget.LinearLayout;

	/// <summary>
	/// This class is a LinearLayout which displays its children in reverse order when its orientation is
	/// vertical and layoutdirection is RTL
	/// </summary>
	public class ReversableLinearLayout : LinearLayout
	{

		public ReversableLinearLayout(Context context, AttributeSet attrs, int defStyle) : base(context, attrs, defStyle)
		{
		}

		public ReversableLinearLayout(Context context) : base(context)
		{
		}

		public ReversableLinearLayout(Context context, AttributeSet attrs) : base(context, attrs)
		{
		}

		/// <summary>
		/// IMPORTANT
		/// When the orientation is vertical and layoutdirection is RTL this method returns view counting
		/// from the end. For example when index is 0 then last view is returnd.
		/// </summary>
		public override View getChildAt(int index)
		{
			if (Orientation == LinearLayout.VERTICAL && LayoutDirection == ViewGroup.LAYOUT_DIRECTION_RTL)
			{
				index = ChildCount - index - 1;
			}
			return base.getChildAt(index);
		}

	}

}